bots/mainbot/castle.js

```diff
@@ -16,16 +16,26 @@ var castle_locs = [];
 
 
 castle.takeTurn = (self) => {
-	self.log('castle taking turn')
-	
+    self.log('castle taking turn')
+	const visible = self.getVisibleRobots();
+	//self.log('the visbile robots are:' + visible) ;
+    castle.countUnits(self, visible);
+
+    utilities.log(self, `Current unit counts: Castles: ${self.castle_count}, Crusaders: ${self.crusader_count}, Prophets: ${self.prophet_count}, Pilgrims: ${self.pilgrim_count}`)
+
+	if(self.prophet_count >= 5){
```

Comment:
Nitpick: wacky indenting in this section

Reply:
I think this should be fixed in the castle-rework branch. I can pr it once we get this through. Let's handle formatting issues on that pr. Sound good?